pkgs/development/libraries/pmix/default.nix

```diff
@@ -46,6 +52,24 @@ stdenv.mkDerivation rec {
 
   postInstall = ''
     find $out/lib/ -name "*.la" -exec rm -f \{} \;
+
+    moveToOutput "bin/pmix_info" "''${!outputDev}"
+    moveToOutput "bin/pmixcc" "''${!outputDev}"
+    moveToOutput "share/pmix/pmixcc-wrapper-data.txt" "''${!outputDev}"
+
+    # The path to the pmixcc-wrapper-data.txt is hard coded and
+    # points to $out instead of dev. Use wrapper to fix paths.
+    wrapProgram $dev/bin/pmixcc \
+      --set PMIX_INCLUDEDIR $dev/include \
+      --set PMIX_PKGDATADIR $dev/share/pmix
+  '';
+
+  postFixup = ''
+    remove-references-to -t $dev $(readlink -f $out/lib/libpmix.so)
+
```

Comment:
```
❯ strings pmix/lib/libpmix.so | grep eeeeeeeeeeeeeeeee
/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-pmix-5.0.1-dev/include
```

We'll need to find out (and document)

1. What are the implications of masking out this path
2. How do we detect if in future upstream adds more references to `$dev` and we start silently masking them out too

Reply:
1. As far as I can tell the include path is only used to store the build info (used for display purposes). I could not find any indication that the wiped out include/dev path is used for anything else (I will add a comment). Note that this pattern is used in all packages that use an openmpi-like build infrastructure.
2. In case upstream changes something and we accidentally wipe out more than intended, it should manifest itself in build failures of packages that depended on pmix and thus is easy to detected. I.e. you will get a error that states something like this `can not find /nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-pmix-5.0.1-dev/include`.